### PR TITLE
Add links and removed dead links

### PR DESCRIPTION
**Pull request by MathewtheCoder**
I have added new links for angular,clojure,r programming and ruby.
I have corrected the link for lua
I removed certain links of android which were dead or outdated

### CODE REVIEW

File: free-courses-en.md [hunk 1]
```diff
@@ -210,6 +213,7 @@
 
 ### Ruby
 
+* [Learn Ruby the Hard Way](http://learnrubythehardway.org/book/)
```
Comment (vhf) [inline free-courses-en.md]: Could you please move this link to `free-programming-books.md`?

File: free-courses-en.md [hunk 2]
```diff
@@ -79,11 +79,13 @@
 
 ### Clojure
 
+* [BraveClojure](http://www.braveclojure.com)
```
Comment (vhf) [inline free-courses-en.md]: Is this a course or a book? If it's a book, could you please move this link to `free-programming-books.md` and directly link to the book?

File: free-courses-en.md [hunk 3]
```diff
@@ -57,6 +55,7 @@
 
 ### AngularJS
 
+* [AngularJS](http://www.angularjsbook.com)
```
Comment (vhf) [inline free-courses-en.md]: Could you please move this link to `free-programming-interactive-tutorials.md`?

**Comment by MathewtheCoder** [comment]
@vhf Can codeacademy or codeschool courses be included in this file or should it be in free-programming-interactive-tutorials.md

**Comment by vhf** [comment]
@MathewtheCoder I'm not sure, I'm not so familiar with these websites. If it's mainly "classic" educational content or if they are video lectures, it's more of a "course". If it's really interactive, for example if you have to actually type code to progress, then it's interactive. If it's a book, then we also have files for these. :)

File: free-courses-en.md (hunk 2 above)
Comment (switchhax) [inline free-courses-en.md]: Direct link to the book: `http://www.braveclojure.com/clojure-for-the-brave-and-true/` ?

File: free-courses-en.md (hunk 1 above)
Comment (switchhax) [inline free-courses-en.md]: Is already there. We have as far as i know all books of the "Learn xy the hard way"-franchise

Comment (vhf) [inline free-courses-en.md]: Good, thanks @switchhax. There no point in duplicating content. @MathewtheCoder could you remove this link?

File: free-courses-en.md (hunk 2 above)
Comment (vhf) [inline free-courses-en.md]: Yep, perfect. Please use this one @MathewtheCoder, but in `free-programming-books.md` instead of here. :)

**Pull request closed without merging**